File: 129. Sum Root to Leaf Numbers/Java/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int sumNumbers(TreeNode root) {
        class Utils {
            int numbers(TreeNode root, int prevSum) {
                if (root == null) {
                    return 0;
                }
                prevSum = prevSum * 10 + root.val;
                if (root.left == null && root.right == null) {
                    return prevSum;
                }
                return numbers(root.left, prevSum) + numbers(root.right, prevSum);
            }
        }
        Utils utils = new Utils();
        return utils.numbers(root, 0);
    }
}

File: 129. Sum Root to Leaf Numbers/Python/Solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution(object):
    def sumNumbers(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        def numbers(root, prev_sum):
            if not root:
                return 0
            prev_sum = prev_sum * 10 + root.val
            if not root.left and not root.right:
                return prev_sum
            return numbers(root.left, prev_sum) + numbers(root.right, prev_sum)

        return numbers(root, 0)
